core/types/calls.cc

```diff
@@ -46,6 +46,16 @@ DispatchResult OrType::dispatchCall(const GlobalState &gs, const DispatchArgs &a
     categoryCounterInc("dispatch_call", "ortype");
     auto leftRet = left.dispatchCall(gs, args.withSelfRef(left));
     auto rightRet = right.dispatchCall(gs, args.withSelfRef(right));
+
+    auto leftOk = leftRet.main.method.exists();
+    auto rightOk = rightRet.main.method.exists();
+    // If only one side is missing the method, dispatch to that one to force an error
+    if (!leftOk && rightOk) {
+        return leftRet;
+    } else if (leftOk && !rightOk) {
+        return rightRet;
+    }
```

Comment:
Nit: can you please write this as:

```c++
if (!leftOk && rightOk) {
  return leftRet;
}
if (leftOk && !rightOk) {
  return rightRet;
}
```

(else after return is unnecessary)

Reply:
Thanks for looking into this.

This solution looks kind of like a hack.

Can you say more about how the code worked and what the problem was? I think that we're likely treating the symptoms rather than treating the cause with this fix.

Reply:
@froydnj: I don't feel strongly about that, but even if it's unnecessary for a machine to have the `else`, I find it way easier for humans to read with the extra `else`.

Reply:
@jez sure! Here's what I did:

First, I debugged things to see if the error was being detected and correctly added to the `DispatchResult` here:

https://github.com/sorbet/sorbet/blob/a9063fd1745324b31b2dcff064194bf2a5c50091/core/types/calls.cc#L554-L561

After I validated that, I looked at how it works for `T.all` here:

https://github.com/sorbet/sorbet/blob/a9063fd1745324b31b2dcff064194bf2a5c50091/core/types/calls.cc#L76-L102

And followed the pattern there, adapting to `Or`, instead of `And`